src/control/cmd/dmg/pretty/storage_query.go

```diff
@@ -44,19 +52,245 @@ func PrintHostStorageUsageMap(hsm control.HostStorageMap, out io.Writer) error {
+
+		ncs := storage.NvmeDevices
+		row[nvmeTitle] = humanize.Bytes(ncs.Total())
+		nvmeFree := ncs.Free()
+		row[nvmeFreeTitle] = humanize.Bytes(nvmeFree)
+		row[nvmeUsageTitle] = common.PercentageString(ncs.Total()-nvmeFree, ncs.Total())
+
 		table = append(table, row)
 	}
 
 	tablePrint.Format(table)
+	return
+}
+
+const (
+	metaRole  = storage.BdevRoleMeta
+	dataRole  = storage.BdevRoleData
+	rankTitle = "Rank"
+)
+
+// Return role combinations for each tier that contains either a meta or data role.
+func getTierRolesForHost(nvme storage.NvmeControllers, metaRolesLast, dataRolesLast *storage.BdevRoles) error {
+	roles := make(map[int]*storage.BdevRoles)
+	for _, c := range nvme {
+		if c.Roles().HasMeta() {
+			if _, exists := roles[metaRole]; !exists {
+				roles[metaRole] = c.Roles()
+			}
+		} else if c.Roles().HasData() {
+			if _, exists := roles[dataRole]; !exists {
+				roles[dataRole] = c.Roles()
+			}
+		}
+	}
+
+	if roles[metaRole].IsEmpty() {
+		return errNoMetaRole
+	}
+
+	if !metaRolesLast.IsEmpty() {
+		// Indicates valid "last" values exist so check consistency.
+		if *roles[metaRole] != *metaRolesLast {
+			return errInconsistentRoles
+		}
+		if roles[dataRole].IsEmpty() {
+			if !dataRolesLast.IsEmpty() {
+				return errInconsistentRoles
+			}
+		} else {
+			if *roles[dataRole] != *dataRolesLast {
+				return errInconsistentRoles
+			}
+		}
+	} else {
+		*metaRolesLast = *roles[metaRole]
+		if !roles[dataRole].IsEmpty() {
+			*dataRolesLast = *roles[dataRole]
+		}
+	}
+
 	return nil
 }
 
+// Print which roles each tier is assigned, only print tiers with meta or data roles.
+// Currently tier-list hardcoded to (META/DATA) but this can be extended.
+func printTierRolesTable(hsm control.HostStorageMap, out, dbg io.Writer) ([]storage.BdevRoles, error) {
+	tierTitle := "Tier"
+	rolesTitle := "Roles"
+
+	tablePrint := txtfmt.NewTableFormatter(tierTitle, rolesTitle)
+	tablePrint.InitWriter(out)
+	table := []txtfmt.TableRow{}
+
+	// Currently only tiers with meta and data are of interest so select implicitly.
+	var metaRoles, dataRoles storage.BdevRoles
+	for _, key := range hsm.Keys() {
+		err := getTierRolesForHost(hsm[key].HostStorage.NvmeDevices, &metaRoles, &dataRoles)
+		if err != nil {
+			hSet := hsm[key].HostSet
+			fmt.Fprintf(dbg, "scan resp for hosts %q: %+v\n", hSet, hsm[key].HostStorage)
+			return nil, errors.Wrapf(err, "hosts %q", hSet)
+		}
+	}
+
+	if metaRoles.IsEmpty() {
+		fmt.Fprintf(dbg, "scan resp: %+v\n", hsm)
+		return nil, errInsufficientScan
+	}
+	table = append(table, txtfmt.TableRow{
+		tierTitle:  "T1",
+		rolesTitle: metaRoles.String(),
+	})
+	rolesToShow := []storage.BdevRoles{metaRoles}
+
+	// Print data role row if assigned to a separate tier from meta role.
+	if !dataRoles.IsEmpty() {
+		table = append(table, txtfmt.TableRow{
+			tierTitle:  "T2",
+			rolesTitle: dataRoles.String(),
+		})
+		rolesToShow = append(rolesToShow, dataRoles)
+	}
```

Comment:
A more generic approach to this could be:
```go
tierRoles := []storage.BdevRoles{metaRoles}
if !dataRoles.IsEmpty() {
        tierRoles = append(tierRoles, dataRoles)
}
for i, roles := range tierRoles {
        table = append(table, txtfmt.TableRow{
		tierTitle:  fmt.Sprintf("T%d", i),
		rolesTitle: roles.String(),
	})
	rolesToShow = append(rolesToShow, roles)
}
```

Reply:
sure